solutions/cf-1225D.mdx

```diff
@@ -0,0 +1,101 @@
+---
+id: cf-1225D
+source: CF
+title: Power Products
+author: Jesse Choe
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/89810)
+
+
+**Time Complexity:** $\mathcal{O}(N\log ^2N \sqrt N)$
+
+<LanguageSection>
+<CPPSection>
+
+## C++ Implementation
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+using ll = long long;
+using pi = pair<int, int>;
+
+int num_of_integers, exponent;
+
+/*
+ * We can define a prime factorization pair as the following:
+ * (a, b) represents a prime number (a) to the power of (b). However, 
+ *  to help us later, we can write (a, b) as (a, b mod k).
+ */
+
+vector<pi> factor(int number){
+	map<int, int> factors;
+	// Prime factorization in O(sqrt(n)) time
+	for(int i = 2; i * i <= number; i++){
+		while(number % i == 0){
+			number /= i;
+			factors[i]++;
+		}
+	}
+	if(number > 1){
+		factors[number]++;
+	}
+	vector<pi> prime_factorization;
+	for(pi primes: factors){
+		if(primes.second % exponent > 0){
+			prime_factorization.push_back({primes.first, primes.second % exponent});
+		}
+	}
+	return prime_factorization;
+}
+
+/*
+ * Let's define the inverse of a prime factorization pair as follows:
+ * Given the pair (a, b), where a represents the prime number and
+ * b represents the exponent mod k, the inverse of (a, b) would be
+ * precisely (a, k - b) since a^b * a^(k - b) = a^k.
+ */
+
+vector<pi> find_factor_inverse(vector<pi> original){
+	int number_of_primes = original.size();
+	vector<pi> inverse(num_of_integers);
+	for(int i = 0; i < number_of_primes; i++){
+		inverse[i].first = original[i].first;
+		inverse[i].second = exponent - original[i].second;
+	}
+	return inverse;
+}
+
+int main(){
+	cin >> num_of_integers >> exponent;
```

Comment:
look i know i said explicit variable names were cool but maybe `num_amt` would be better
or even just `size`